### PR TITLE
feat: Drops requiring funds to enable auto-teleport

### LINKED ISSUE #8482: Drops requiring funds to enable auto-teleport
context: 

>if address has balance on relay chain in DOTs, make ability to auto-teleport this funds for listing as if someone uses on->ramp, funds will arrive on relay chain anyway, not on Assets Hub 

>image

# Design 
- updated file in confirmation modals 
https://www.figma.com/file/JXNtryaNYhY0XJNArzvPQx/Handoff?type=design&node-id=1458-19091&mode=design&t=24r41xUB3tUOH8bw-11
image
- for auto teleport to work, the user has to have 1.2 DOT on polkadot 
- this way we can enable auto teleport and have user with funds on Polkadot asset hub
- added chain next to the button (I have also changed the copy inside the modal a bit) 
image

Let me now if anything is not clear. I will try to explain asap

### LINKED ISSUE #8526: Check ED on auto teleport
Since we have a loading modal, there is time to check if the auto teleport will lower the account - from which the asset is bridge - under its existential deposit. If that happens, let's not enable the auto teleport at all 
- we should prevent users losing their funds because of that. 

### rel 
- https://github.com/kodadot/nft-gallery/issues/8359
- https://github.com/kodadot/nft-gallery/pull/8505

### PR DESCRIPTION
**Pull request by hassnian**
## PR Type

- [ ] Bugfix
- [x] Feature
- [ ] Refactoring

## Context

- [x] Closes #8482
- [x] Closes #8526

#### Did your issue had any of the "$" label on it?

- [x] Fill up your DOT address: [Payout](https://kodadot.xyz/dot/transfer?target=13QUj3pZyFNfYj4AM336hRdyLQbevj5H3sR4PKmLEXLdwZhh)

#### Community participation

- [x] [Are you at KodaDot Discord?](https://discord.gg/35hzy2dXXh)

## Screenshot 📸

- [ ] My fix has changed **something** on UI; 

can autoteleport

https://github.com/kodadot/nft-gallery/assets/44554284/1e277d46-9463-4fb6-a0b1-033b7543897b

auto teleport not available

https://github.com/kodadot/nft-gallery/assets/44554284/e2c32e8e-2de9-4813-a46d-851819e91524

### <samp>🤖[[deprecated]](https://githubnext.com/copilot-for-prs-sunset) Generated by Copilot at 0bda4cc</samp>

Improved the auto-teleport feature by using a utility function to get the existential deposit for any chain. Refactored the code in `useAutoTeleportTransitionDetails.ts` and `teleport.ts` to use this function.

### <samp>🤖[[deprecated]](https://githubnext.com/copilot-for-prs-sunset) Generated by Copilot at 0bda4cc</samp>

> _`getChainExistentialDeposit`_
> _Calculates balance for any chain_
> _No more hardcoded values_

### CODE REVIEW

**Comment by exezbcz** [comment]
maybe change button label to "Continue"

image

in this state when everything is done

**Comment by exezbcz** [comment]
and also small detail, I am certain its not coming from this pr, but could you please fix the stroke on the context connected with message 
image

- it should be grey as before

File: locales/en.json [hunk 1]
```diff
@@ -1650,6 +1652,11 @@
         "title": "Minting NFT Collection",
         "submit": "Minting NFT Collection..."
       },
+      "drop": {
+        "confirm": "Teleport Required Amount",
+        "action": "Claim free",
+        "item": "NFT"
```
Comment (roiLeo) [inline locales/en.json:1658]: already exist

File: locales/en.json [hunk 2]
```diff
@@ -1650,6 +1652,11 @@
         "title": "Minting NFT Collection",
         "submit": "Minting NFT Collection..."
       },
+      "drop": {
+        "confirm": "Teleport Required Amount",
+        "action": "Claim free",
```
Comment (roiLeo) [inline locales/en.json:1657]: always free?

File: locales/en.json [hunk 3]
```diff
@@ -1650,6 +1652,11 @@
         "title": "Minting NFT Collection",
         "submit": "Minting NFT Collection..."
       },
+      "drop": {
+        "confirm": "Teleport Required Amount",
```
Comment (roiLeo) [inline locales/en.json:1656]:
```suggestion
        "confirm": "Teleport required amount",
```

**Review by roiLeo — APPROVED** [review]

**Comment by prury** [comment]
@hassnian 

https://github.com/kodadot/nft-gallery/assets/36627808/60390d20-72a0-4475-aef0-1034815badf5

- Complete all required steps button is getting active too soon - around 1:10
- When i cancelled the mint it showed the button to teleport again, maybe because funds were not updated? but soon after it worked well (not sure if it is a problem tho) - around 1:40

**Comment by exezbcz** [comment]
@prury uf, 30 seconds to check the funds, does the pr of modal slowness solve this issue or?

**Comment by prury** [comment]
> @prury uf, 30 seconds to check the funds, does the pr of modal slowness solve this issue or?

everything i tested there so far is faster, i hope it does! but balances have their ups and downs as well, some days they load super fast, sometimes they are slow 😄

File: locales/en.json (hunk 3 above)
Comment (hassnian) [reply locales/en.json:1656]: fixed

File: locales/en.json (hunk 1 above)
Comment (hassnian) [reply locales/en.json:1658]: wdym

File: locales/en.json (hunk 2 above)
Comment (hassnian) [reply locales/en.json:1657]: aren't all drops that we are doing atm free?

maybe it should be generative_drop , what do you think ?

**Comment by daiagi** [comment]
>some days they load super fast, sometimes they are slow 😄

sometimes I'm slow, sometimes I'm not, sometimes I'm slow
hello?
https://www.youtube.com/watch?v=lwLLFbC1H0c

**Comment by hassnian** [comment]
@prury @exezbcz  fixed 

check again thanks 

https://github.com/kodadot/nft-gallery/assets/44554284/f098d777-8910-4bd0-a0cb-7d3fa9687581

File: locales/en.json (hunk 1 above)
Comment (roiLeo) [reply locales/en.json:1658]: value `NFT` already exist in translation file no need to duplicate another key value

File: locales/en.json (hunk 2 above)
Comment (roiLeo) [reply locales/en.json:1657]: not sure about that one, yes for now they are free

Comment (hassnian) [reply locales/en.json]: checked current implementation and its not being checked

dropcard.vue

CleanShot 2023-12-16 at 15 54 25@2x

File: locales/en.json (hunk 1 above)
Comment (hassnian) [reply locales/en.json]: fixed pls review

**Comment by hassnian** [comment]
added autotelport to holder of 

@vikiival meta should be less right ?
CleanShot 2023-12-16 at 16 46 17@2x

CleanShot 2023-12-16 at 16 45 31@2x

checked the code and it looks ok that value is the issue

**Comment by yangwao** [comment]
@hassnian let's resolve conflicts and we can try push it live

**Comment by yangwao** [comment]
Thanks!
pay 50 usd

**Comment by yangwao** [comment]
😍 Perfect, I’ve sent the payout 
💵 $50 @ 7.17 USD/DOT ~ 6.974 $DOT
🧗 [13QUj3pZyFNfYj4AM336hRdyLQbevj5H3sR4PKmLEXLdwZhh](https://polkadot.subscan.io/account/13QUj3pZyFNfYj4AM336hRdyLQbevj5H3sR4PKmLEXLdwZhh)
🔗 [0xa6dd77dc3dbfb014d911a8f69bf25d8833abe4c9235953a65ece738fa1131844](https://polkadot.subscan.io/extrinsic/0xa6dd77dc3dbfb014d911a8f69bf25d8833abe4c9235953a65ece738fa1131844)

🪅 Let’s grab another issue and get rewarded! 
🪄 [github.com/kodadot/nft-gallery/issues](https://github.com/kodadot/nft-gallery/issues/)

**Comment by yangwao** [comment]
Seems we needs more testing here, introduced this bug 
- https://github.com/kodadot/nft-gallery/issues/8619

**Comment by vikiival** [comment]
> meta should be less right ?



image

**Pull request merged**